lib/cocoapods/sandbox/headers_store.rb

```diff
@@ -24,21 +24,34 @@ def initialize(sandbox, relative_path)
         @sandbox       = sandbox
         @relative_path = relative_path
         @search_paths  = []
+        @search_paths_cache = {}
+        @search_paths_key = Struct.new(:platform_name, :target_name)
       end
 
       # @param  [Platform] platform
       #         the platform for which the header search paths should be
       #         returned
       #
+      # @param  [String] target_name
+      #         the target for which the header search paths should be
+      #         returned, may be `nil`.
+      #
       # @return [Array<String>] All the search paths of the header directory in
       #         xcconfig format. The paths are specified relative to the pods
       #         root with the `${PODS_ROOT}` variable.
       #
-      def search_paths(platform)
-        platform_search_paths = @search_paths.select { |entry| entry[:platform] == platform.name }
-
+      def search_paths(platform, target_name = nil)
+        key = @search_paths_key.new(platform.name, target_name)
+        if @search_paths_cache.key?(key)
+          return @search_paths_cache[key]
+        end
+        platform_search_paths = @search_paths.select do |entry|
+          matches_platform = entry[:platform] == platform.name
+          next matches_platform unless target_name
```

Comment:
NIT: This would be more clear as:
```
matches_platform = entry[:platform] == platform.name
matches_target = target_name.nil? || (entry[:path].basename.to_s == target_name)
matches_platform && matches_target
```

Reply:
fixed.